src/V3Randomize.cpp

```diff
@@ -201,29 +221,33 @@ class ConstraintExprVisitor final : public VNVisitor {
     // VISITORS
     void visit(AstNodeVarRef* nodep) override {
         if (editFormat(nodep)) return;
+        auto classp = nodep->classOrPackagep();
         // In SMT just variable name, but we also ensure write_var for the variable
         const std::string smtName = nodep->name();  // Can be anything unique
         nodep->replaceWith(new AstSFormatF{nodep->fileline(), smtName, false, nullptr});
 
         AstVar* const varp = nodep->varp();
+        AstVarScope* const varScopep = nodep->varScopep();
 
         VL_DO_DANGLING(pushDeletep(nodep), nodep);
 
-        if (!varp->user4()) {
-            varp->user4(true);
-            AstCMethodHard* const methodp = new AstCMethodHard{
-                varp->fileline(), new AstVarRef{varp->fileline(), m_genp, VAccess::READWRITE},
-                "write_var"};
-            methodp->dtypeSetVoid();
-            methodp->addPinsp(new AstVarRef{varp->fileline(), varp, VAccess::WRITE});
-            methodp->addPinsp(new AstConst{varp->dtypep()->fileline(), AstConst::Unsized64{},
-                                           (size_t)varp->width()});
-            AstNodeExpr* const varnamep
-                = new AstCExpr{varp->fileline(), "\"" + smtName + "\"", varp->width()};
-            varnamep->dtypep(varp->dtypep());
-            methodp->addPinsp(varnamep);
-            m_taskp->addStmtsp(methodp->makeStmt());
-        }
+        varp->user4(true);
+        auto genRefp = new AstVarRef{varp->fileline(), m_genp, VAccess::READWRITE};
```

Comment:
```suggestion
        AstVarRef* const genRefp = new AstVarRef{varp->fileline(), m_genp, VAccess::READWRITE};
```

Please check all code below for "const" missing.  (Hint - another one a few lines down ;)

Reply:
Done.